coordinator/src/main/scala/filodb.coordinator/ShardAssignmentStrategy.scala

```diff
@@ -88,31 +102,53 @@ class K8sStatefulSetShardAssignmentStrategy extends ShardAssignmentStrategy with
           val unassignedShardSet = mapper.unassignedShards.toSet
           val potentialShardsMappedToThisCoordinator =
             (firstShardThisNode until firstShardThisNode + numShardsThisHost).toList.
-              filter(unassignedShardSet.contains(_))
+              filter(unassignedShardSet.contains)
 
           if(potentialShardsMappedToThisCoordinator.nonEmpty) {
             logger.info("Using hostname resolution for shard mapping, mapping host={} to shards={}",
               hostName, potentialShardsMappedToThisCoordinator)
           }
           potentialShardsMappedToThisCoordinator
-        case None                      =>
-          // Host name does not have the ordinal at the end like a stateful set needs to have, delegate to default
-          //strategy
-          logger.info(
-            "Falling back to DefaultShardAssignmentStrategy as hostname does not match k8s StatefulSet convention" +
-              " for coordinator {}", coord.path.address.host.getOrElse(InetAddress.getLocalHost.getHostName))
-          DefaultShardAssignmentStrategy.shardAssignments(coord, dataset, resources, mapper)
-      }
-
+        case Some((hostName, _))       => retryWithBackoff(coord, dataset, resources, mapper,
+                                            hostName, currentAttempt, maxAssignmentAttempts)(shardAssignmentsWithRetry)
+        case None                      => retryWithBackoff(coord, dataset, resources, mapper, hostName = "",
+                                            currentAttempt, maxAssignmentAttempts)(shardAssignmentsWithRetry)
+  }
+  // scalastyle:on method.length
+
+  private def retryWithBackoff[T](coord: ActorRef, dataset: DatasetRef, resources: DatasetResourceSpec,
+                               mapper: ShardMapper, hostName: String,
+                               currentAttempt: Int, maxAssignmentAttempts: Int)
+                              (f: (ActorRef, DatasetRef, DatasetResourceSpec, ShardMapper, Int, Int) => T): T= {
+    // Host name does not have the ordinal at the end like a stateful set needs to have, retry we we have
+    // retries left
+    if (currentAttempt > maxAssignmentAttempts) {
+      logger.error(s"Unable to resolve host names after $currentAttempt attempts, terminating now")
+      sys.exit(1)
+    } else {
+      val retryAfter = 1 << currentAttempt
```

Comment:
Doesn't linear backoff work better here ? DNS lookup resulting in a denial of service is less likely. We want the issue to resolve asap.

Reply:
The reason I kept exponential backoff is we don't need to configure the fixed interval. It will try in 2, 4, 8, 16, 32, 64.. seconds and only number of attempts is to be configured. What can be done is cap this retry interval to a fixed hardcoded number like 60 seconds.